main.rb

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env ruby
+
+require './lib/app'
+
+def age
+  print 'Age: '
+  age = gets.chomp
+  if age == ''
+    puts 'Age cannot be blank'
+    age
+  end
+  age
+end
+
+def name
+  print 'Name: '
+  name = gets.chomp
+  name = nil if name == ''
+  name
+end
+
+def date
+  print 'Date (yy/mm/dd): '
+  date = gets.chomp
+  if date == ''
+    puts 'A date is required!'
+    date
+  end
+  date
+end
+
+def permission
+  print 'Has parent permission? [Y/N]: '
+  permission = gets.chomp
+  /y/i.match?(permission) == true
+end
+
+def specialization
+  print 'Specialization: '
+  gets.chomp
+end
+
+def create_person(app)
+  print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+  person = gets.chomp
+  if person == '1' # create student
+    age = age()
+    name = name()
+    permission = permission()
+
+    app.create_student(age, nil, name, parent_permission: permission)
+  elsif person == '2' # create teacher
+    age = age()
+    name = name()
+    specialization = specialization()
+
+    app.create_teacher(age, specialization, name)
+  else
+    puts 'You need to select an actual number'
+    start(app)
+  end
+  puts 'Person created succesfully'
+end
+
+def create_book(app)
+  print 'Title: '
+  title = gets.chomp
+  print 'Author: '
+  author = gets.chomp
+
+  app.create_book(title, author)
+  puts 'Book created succesfully'
+end
+
+def display_persons(persons)
+  persons.each_with_index do |person, index|
+    result = "#{index}) "
+    result += "[#{person.class.name}] "
+    result += "Name: #{person.name.capitalize}, "
+    result += "ID: #{person.id}, "
+    result += "Age: #{person.age}"
+    puts result
+  end
+end
+
+def display_books(books)
+  books.each_with_index do |book, index|
+    result = "#{index}) "
+    result += "Title: \"#{book.title.capitalize}\", "
+    result += "Author: #{book.author.capitalize}"
+    puts result
+  end
+end
+
+def start_rental(app)
+  books = app.books
+  persons = app.people
+  if books.empty? && persons.empty?
+    puts 'Ensure you have books in library and people registered!'
+  else
+    puts 'Select a book from the following list by number'
+    display_books(books)
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by numnber (not id)'
+    display_persons(persons)
+
+    person_index = gets.chomp.to_i
+
+    date = date()
+
+    app.create_a_rental(date, app.books[book_index], app.people[person_index])
+    puts 'Rental created succesfully!'
+  end
+end
+
+def display_rentals(app)
+  print 'ID of person: '
+  id = gets.chomp.to_i
+  app.list_person_rentals(id)
+  start(app)
+end
+
```

Comment:
- [ ] Kindly move this part of the app logic into app,rb, as the main,rb file is meant for presenting the initial user inputs and for starting the app.